src/fios.cpp

```diff
@@ -205,12 +205,16 @@ const char *FiosBrowseTo(const FiosItem *item)
  */
 static void FiosMakeFilename(char *buf, const char *path, const char *name, const char *ext, const char *last)
 {
-	const char *period;
+	const char *buf_start = buf;
+	buf = strecpy(buf, path, last);
+	/* Remove trailing path separator, if present */
+	if (buf > buf_start && !IsUtf8Part(buf[-1]) && buf[-1] == PATHSEPCHAR) buf--;
```

Comment:
I've add an out-of-bounds check, and also fix the code to support international paths.